docs: slim bindings v0.6.0 (#253)
agntcy/docs: docs/messaging/slim-group-tutorial.md

                f"{shared_session_container[0].src} > "
            )

            if user_input.lower() in ("exit", "quit"):
                # Also terminate the receive loop.
                await local_app.delete_session(shared_session_container[0])
                break

            try:
                # Send message to the channel_name specified when creating the session.
                # As the session is multicast, all participants will receive it.
                # calling publish_with_destination on a multicast session will raise an error.
                await shared_session_container[0].publish(user_input.encode())
            except KeyboardInterrupt:
                # Handle Ctrl+C gracefully
                break
            except Exception as e:
                print_formatted_text(f"-> Error sending message: {e}")
    except asyncio.CancelledError:
        # Handle task cancellation gracefully
        pass
```

Messages are sent using `shared_session_container[0].publish(user_input.encode())`.
Only the payload is provided and there is no explicit destination, because the
multicast channel was fixed at session creation and delivery fan-outs to all
participants.

> ### Run the Gruop Comminication Example

Reviewer: ```suggestion
### Run the Gruop Communication Example
```

In this example we use two participants: `agntcy/ns/client-1` and `agntcy/ns/client-2`.
Authentication uses a shared secret. In the SLIM repository, go to the folder 
`slim/data-plane/python/bindings/examples` and run these commands in two different terminals:

```bash
uv run --package slim-bindings-examples multicast                               \
    --local agntcy/ns/client-1                                                  \
    --slim '{"endpoint": "http://localhost:46357", "tls": {"insecure": true}}'  \
    --shared-secret "secret"
```
```bash
uv run --package slim-bindings-examples multicast                               \
    --local agntcy/ns/client-2                                                  \
    --slim '{"endpoint": "http://localhost:46357", "tls": {"insecure": true}}'  \
    --shared-secret "secret"

```
This will start two participants authenticated with a shared secret.
The outcome of these commands should look like:

```bash
Warning: Falling back to shared-secret authentication. Don't use this in production!
Agntcy/ns/client-1/456243414154990054        Created app
Agntcy/ns/client-1/456243414154990054        Connected to http://localhost:46357

> -> Waiting for session...

Reviewer: We removed the -> from the output

    --remote agntcy/ns/chat                                                     \
    --invites agntcy/ns/client-1                                                \
    --invites agntcy/ns/client-2                                                \
    --enable-mls
```

The result should look like:

```bash
Agntcy/ns/moderator/16858445264489265394     Created app
Agntcy/ns/moderator/16858445264489265394     Connected to http://localhost:46357
Creating new multicast session (moderator)... 169ca82eb17d6bc2/eef9769a4c6990d1/fc9bbc406957794b/ffffffffffffffff (agntcy/ns/moderator/ffffffffffffffff)
agntcy/ns/moderator -> add 169ca82eb17d6bc2/eef9769a4c6990d1/58ec40d7c837e0b9/ffffffffffffffff (agntcy/ns/client-1/ffffffffffffffff) to the group
agntcy/ns/moderator -> add 169ca82eb17d6bc2/eef9769a4c6990d1/b521a3788f1267a8/ffffffffffffffff (agntcy/ns/client-2/ffffffffffffffff) to the group
Welcome to the group 169ca82eb17d6bc2/eef9769a4c6990d1/4abb367236cabc2a/ffffffffffffffff (agntcy/ns/chat/ffffffffffffffff)!
Send a message to the group, or type 'exit' or 'quit' to quit.
169ca82eb17d6bc2/eef9769a4c6990d1/fc9bbc406957794b/e9f53aa5ef3fb8f2 (agntcy/ns/moderator/e9f53aa5ef3fb8f2) >
```

> Now `client-1` and `client-2` are invited to the gruop so on both of them you should 

Reviewer: ```suggestion
Now `client-1` and `client-2` are invited to the group so on both of them you should 
```

be able to see a welcome message such as:

```bash
Welcome to the group 169ca82eb17d6bc2/eef9769a4c6990d1/4abb367236cabc2a/ffffffffffffffff (agntcy/ns/chat/ffffffffffffffff)!
Send a message to the group, or type 'exit' or 'quit' to quit.
169ca82eb17d6bc2/eef9769a4c6990d1/58ec40d7c837e0b9/6a34b65ebc955471 (agntcy/ns/client-1/6a34b65ebc955471) >
```

At this point, you can write messages from any terminal and they will be received by all the other group participants.

## Group Communication Using the SLIM Controller

Previously, we saw how to run group communication using the Python bindings with an in‑application moderator. 

> This particular participant has to create the mulitcast session and invite all the other participants.

Reviewer: ```suggestion
This particular participant has to create the multicast session and invite all the other participants.
```

In this section, we describe how to create and orchestrate a group using the SLIM Controller and we show how all
this functions can be delegated to the controller. We reuse the same multicast example code also in this section.

Identity handling is unchanged between the two approaches; refer back to [SLIM Identity](#configure-client-identity-and-implement-the-slim-app). Below are the steps to run the controller-managed version.

### Application Differencies

With the controller, you do not need to set up a moderator in your application. All participants can be run as we did for `client-1` and `client-2` in the previous examples. In code, this means you can avoid to create a new multicast session (using `local_app.create_session`) and the invitation loop. You only need to implement the `receive_loop` where the application waits for new sessions. This greatly simplifies your code.

### Run the Gruop Communincation example

> Now we will show how to setupa gruop using the SLIM Contoreller. The reference code for the 

Reviewer: ```suggestion
Now we will show how to setup a group using the SLIM Controller. The reference code for the 
```

> applciation is still [multicast.py](https://github.com/agntcy/slim/blob/main/data-plane/python/bindings/examples/src/slim_bindings_examples/multicast.py). To run this example, follow the steps listed here.

Reviewer: ```suggestion
application is still [multicast.py](https://github.com/agntcy/slim/blob/main/data-plane/python/bindings/examples/src/slim_bindings_examples/multicast.py). To run this example, follow the steps listed here.
```

```bash
cat << EOF > ./config.yaml
northbound:
  httpHost: localhost
  httpPort: 50051

southbound:
  httpHost: localhost
  httpPort: 50052

reconciler:
  threads: 3

logging:
  level: DEBUG
EOF
```

This config defines two APIs exposed by the controller:
- Northbound API: used by an operator (e.g. via slimctl) to configure channels and participants, as well as the SLIM network.
- Southbound API: used by SLIM nodes to synchronize with the controller.

Start the controller with Docker:

```bash
docker run -v ./config.yaml:/config.yaml \
    ghcr.io/agntcy/slim/control-plane:latest \
    --config /config.yaml
```

If everything goes fine, you should see an output like this one:

```bash
2025-10-06T08:06:06Z INF Starting route reconcilers
2025-10-06T08:06:06Z INF Starting Route Reconciler thread_name=reconciler-1
2025-10-06T08:06:06Z INF Starting Route Reconciler thread_name=reconciler-0
2025-10-06T08:06:06Z INF Starting Route Reconciler thread_name=reconciler-2
2025-10-06T08:06:06Z INF Southbound API Service is Listening on 127.0.0.1:50052
2025-10-06T08:06:06Z INF Northbound API Service is listening on 127.0.0.1:50051
```

#### Run the SLIM Node

> With the controller running, start a SLIM node configured to talk to it over the Southbound API. This node config includes two additional features compared to the file proposed in the previous sessions:

Reviewer: ```suggestion
With the controller running, start a SLIM node configured to talk to it over the Southbound API. This node config includes two additional settings compared to the file proposed in the previous sessions:
```

2025-10-06T08:22:54.530157Z  INFO main ThreadId(01) application_lifecycle: slim_service::service: starting service
2025-10-06T08:22:54.530193Z  INFO main ThreadId(01) application_lifecycle: slim_service::service: starting server 0.0.0.0:46357
...
```

On the Controller side, you can see that the new node registers itself to the controller. The
output should be similar to the following one:
```bash
2025-10-06T11:47:14+02:00 INF Registering node with ID: slim/0 svc=southbound
2025-10-06T11:47:14+02:00 INF Connection details: [endpoint: 127.0.0.1:46357] svc=southbound
2025-10-06T11:47:14+02:00 INF Create generic routes for node node_id=slim/0 service=RouteService
2025-10-06T11:47:14+02:00 INF Sending routes to registered node slim/0 node_id=slim/0
2025-10-06T11:47:14+02:00 INF Sending configuration command to registered node connections_count=0 message_id=8e9d311a-0012-4fb2-93dc-cda2cb0dd2ef node_id=slim/0 subscriptions_count=0 subscriptions_to_delete_count=0
2025-10-06T11:47:14+02:00 INF Sending routes completed successfully ack_messages=[] node_id=slim/0 original_message_id=8e9d311a-0012-4fb2-93dc-cda2cb0dd2ef
```

#### Run the Participants

> Because the controller manages the group lifecycle, no participant needs to be designated as moderator in code. Every application instance just waits for a session invite. In three separate terminals, strating from the folder 

Reviewer: ```suggestion
Because the controller manages the group lifecycle, no participant needs to be designated as moderator in code. Every application instance just waits for a session invite. In three separate terminals, starting from the folder 
```

The full name of the application can be taken from the output in the console. The value
`9494657801285491688` is the actual id of the `client-1` application returned by
SLIM. In your case, this value will be different.

Expected response from `slimctl`:

```bash
Received response: agntcy/ns/xyIGhc2igNGmkeBDlZ
```

The value `agntcy/ns/xyIGhc2igNGmkeBDlZ` is the channel (or group) identifier (name) that must be used in subsequent commands.

On the application side, `client-1` was addeded to the session so you should see 
something like this:

```bash
Welcome to the group 169ca82eb17d6bc2/eef9769a4c6990d1/e8ab33f6d6111780/ffffffffffffffff (agntcy/ns/xyIGhc2igNGmkeBDlZ/ffffffffffffffff)!
Send a message to the group, or type 'exit' or 'quit' to quit.
169ca82eb17d6bc2/eef9769a4c6990d1/58ec40d7c837e0b9/83c3ccf725835be8 (agntcy/ns/client-1/83c3ccf725835be8) >
```

##### Add Participants

Now that the new group is created, add the additional participants `client-2` and `client-3` using the following `slimctl` commands:

```bash
./slimctl participant add -c agntcy/ns/xyIGhc2igNGmkeBDlZ agntcy/ns/client-2
./slimctl participant add -c agntcy/ns/xyIGhc2igNGmkeBDlZ agntcy/ns/client-3
```

> The xpected `slimctl` output is:

Reviewer: ```suggestion
The expected `slimctl` output is:
```
